src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud.java

```diff
@@ -170,6 +171,9 @@ public class KubernetesCloud extends Cloud implements PodTemplateGroup {
     @CheckForNull
     private GarbageCollection garbageCollection;
 
+    @NonNull
+    private List<KubernetesCloudTrait> traits = new ArrayList<>();
```

Comment:
```suggestion
    private DescribableList<KubernetesCloudTrait, Descriptor<KubernetesCloudTrait>> traits = new DescribableList<>(Saveable.NOOP);
```

would likely allow you to simplify the setter/getter structure. See example in [git-plugin](https://github.com/jenkinsci/git-plugin/blob/343b16782667e8238566a1e32293f45183151a1f/src/main/java/hudson/plugins/git/GitSCM.java#L162).